common/version.go

```diff
@@ -0,0 +1,53 @@
+package common
+
+import (
+	"fmt"
+	"regexp"
+	"runtime/debug"
+	"strings"
+)
+
+type AppVersionInfo struct {
+	Version   string
+	BuildDate string
+}
+
+var AppVersion AppVersionInfo //nolint:gochecknoglobals
+
+// these variables are dynamically set at build.
+var version = "dev"
+var buildDate = "" //nolint:gochecknoglobals
+
+func (v *AppVersionInfo) Shortline() string {
+	if v.BuildDate == "" {
+		return v.Version
+	}
+
+	return fmt.Sprintf("%s (%s)", v.Version, v.BuildDate)
+}
+
+func (v *AppVersionInfo) ChangelogURL() string {
+	path := "https://github.com/dunglas/mercure"
+	r := regexp.MustCompile(`^v?\d+\.\d+\.\d+(-[\w.]+)?$`)
```

Comment:
Maybe can we remove this regex and just check if `version == dev`?

Reply:
The regex has been added to deal with the `--snapshot` flag.

This flag can be used to run a build without create a new tag.
In this case the Github link would not be valid.

But it may not be necessary ?